base/locks.jl

```diff
@@ -0,0 +1,138 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
```

Comment:
Yes. I'm working on the interaction between threads and tasks, which is also needed for I/O. When that is properly designed, these will get merged.